### ISSUE TITLE
Wrongly lint tsconfig.json when using eslint

### ISSUE THREAD

**Issue by Austaras** [TypeStrong/fork-ts-checker-webpack-plugin#312]
## Current behavior
When using eslint, `tsconfig.json` is wrong included and cause lots of error

## Steps to reproduce the issue
passing `{ eslint: true }` to fork-ts-checker-webpack-plugin

## Issue reproduction repository
https://github.com/Austaras/boilerplate

## Environment
 - **fork-ts-checker-webpack-plugin**: 1.4.0
 - **typescript**: 3.5.3
 - **webpack**: 4.35.3

after some digging I found out that also passing `{ useTypescriptIncrementalApi: false }` will solve this problem.

**Comment by Austaras** [comment]
after some digging I found out that also passing `{ useTypescriptIncrementalApi: false }` will solve this problem.

**Comment by Austaras** [comment]
Well, after some more digging, seems like when enable useTypescriptIncrementalApi, this pulgin will get tsconfig.json as changed file at first compile. TSLint will ignore any file that isn't js or ts, while ESLint doesn't and that is the cause of the problem.

**Comment by johnnyreilly** [comment]
Hi @austaras

Thanks for reporting. That's super curious and I'd like to understand why it's happening. I'm hoping you can help me. If you take a look at this minimal example here:

https://github.com/TypeStrong/ts-loader/tree/master/examples/fork-ts-checker-webpack-plugin

The issue does not present itself and it is using `useTypescriptIncrementalApi`. I'd like to understand what's different between your example and this one. Would you be able to take that example and see what triggers the different behaviour please?

I'd love to understand what causes the issues you're seeing anyway. Also, what are the errors you're being presented with? You don't mention in the issue...

Incidentally, we can easily work around this by adding an extra case to the `if` statement [here](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/blob/34b16c5bb9e003b963a22c5c20ae265cbbb77618/src/ApiIncrementalChecker.ts#L204) to check if the file is a `tsconfig.json` and excluding if so. I'd be happy to receive a PR that did that if you wanted to submit one.

```js
if (this.isFileExcluded(updatedFile) || updatedFile.endsWith('tsconfig.json')) {
```

**Comment by Austaras** [comment]
@johnnyreilly the main difference is I use json as config file so I simply pass `"./tsconfig.json"` to `parseOptions -> project` while your example pass `path.resolve(__dirname, './tsconfig.json')`. Change that to `'./tsconfig.json'` in your example will generate same error.

It's true that a dirty and quick fix is possible but I believe in a wide used package like this one such fix is bad for long term maintenance. However I don't know what's the difference between the two value and even if it's possible to use path in json due to my lack of knowledge on ESLint. I hope you will find a better fix. 

BTW I got some quite generic error like expect an expression and should use single quotes

**Comment by johnnyreilly** [comment]
Thanks! 

Given ESLint shouldn't parse JSON anyway, I think my suggestion to exclude `tsconfig.json` is probably fine. I'm on my phone right now, but if you'd like to submit a PR with my suggested fix I'll happily merge it and get this out there.

https://github.com/eslint/eslint/issues/5076

**Comment by johnnyreilly** [comment]
Actually I've got a screen and put together a prospective fix: https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/pull/313

Could you give this PR a try locally and confirm if this resolves your issue please?  Thanks!

**Comment by Austaras** [comment]
It works fine on my project.
